doc/neps/nep-0056-array-api-main-namespace.rst

```diff
@@ -0,0 +1,524 @@
+
+
+New ``unique_*`` functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``unique`` function, with ``return_index``, ``return_inverse``, and
+``return_counts`` arguments that influence the cardinality of the returned
+tuple, is replaced in the Array API by four respective functions:
+``unique_all``, ``unique_counts``, ``unique_inverse``, and ``unique_values``.
+This new API helps to better anticipate the return object and support clearer
+typing.
+
+
+``np.bool`` addition
+^^^^^^^^^^^^^^^^^^^^
+
+One of the aliases that used to live in NumPy but got removed was ``np.bool``.
+To comply with the Array API it got reintroduced with a different meaning, as
+now it points to NumPy's bool instead of a Python builtin. This change is a
+good idea and we were planning to make it anyway, because ``bool`` is a nicer
+name than ``bool_``. However, we may not have scheduled that reintroduction of
+the name for 2.0 perhaps if it had not been part of the array API standard.
+
+
+Related work
+------------
+
+The array API standard (`html docs <https://data-apis.org/array-api/2022.12/>`__,
+`repository <https://github.com/data-apis/array-api/>`__) is the first related
+work; a lot of design discussion in its issue tracker may be relevant in case
+reasons for particular decisions need to be found.
+
+Downstream adoption from array-consuming libraries is actively happening at the moment,
+see for example:
+
+- scikit-learn `docs on array API support <https://scikit-learn.org/dev/modules/array_api.html>`__ and
+  `PRs <https://github.com/scikit-learn/scikit-learn/pulls?q=is%3Aopen+is%3Apr+label%3A%22Array+API%22>`__ and
+  `issues <https://github.com/scikit-learn/scikit-learn/issues?q=is%3Aopen+is%3Aissue+label%3A%22Array+API%22>`__
+  labeled with *Array API*.
+- SciPy `docs on array API support <http://scipy.github.io/devdocs/dev/api-dev/array_api.html>`__
+  and `PRs <https://github.com/scipy/scipy/pulls?q=is%3Aopen+is%3Apr+label%3A%22array+types%22>`__
+  and `issues <https://github.com/scipy/scipy/issues?q=is%3Aopen+is%3Aissue+label%3A%22array+types%22>`__ labeled with *array types*.
+- Einops `docs on supported frameworks <https://einops.rocks/#supported-frameworks>`__
+  and `PR to implement array API standard support <https://github.com/arogozhnikov/einops/pull/261>`__.
+
+Other array libraries either already have support or are implementing support
+for the array API standard (in sync with the changes for NumPy 2.0, since they
+usually try to be as compatible to NumPy as possible). For example:
+
+- CuPy's `docs on array API support <https://docs.cupy.dev/en/stable/reference/array_api.html>`__
+  and `PRs labelled with array-api <https://github.com/cupy/cupy/pulls?q=is%3Aopen+is%3Apr+label%3Aarray-api>`__.
+- JAX: enhancement proposal `Scope of JAX NumPy & SciPy Wrappers <https://jax.readthedocs.io/en/latest/jep/18137-numpy-scipy-scope.html#axis-2-array-api-alignment>`__
+  and `PR with initial implementation <https://github.com/google/jax/pull/16099>`__.
```

Comment:
A more up-to-date reference is JAX's tracking issue for array API support: https://github.com/google/jax/issues/18353